### PR TITLE
Additional routes

### PR DESCRIPTION
**Pull request by s2504s**
# What
* Create different route tables for the each public subnet
* Add ability to create different additional routes into public and private route tables

## Why
* We need this features in order to resolve legacy routes issue
* Enhancement this module

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -46,3 +46,13 @@ variable "nat_gateway_enabled" {
   description = "Flag to enable/disable NAT gateways"
   default     = "true"
 }
+
+variable "additional_routes_public" {
```
Comment (comeanother) [inline variables.tf]: Rename to `additional_public_routes`.

File: variables.tf [hunk 2]
```diff
@@ -46,3 +46,13 @@ variable "nat_gateway_enabled" {
...
+  type    = "list"
+  default = []
+}
+
+variable "additional_routes_private" {
```
Comment (comeanother) [inline variables.tf]: Rename to `additional_private_routes`.

File: private.tf [hunk 3]
```diff
@@ -35,6 +35,13 @@ resource "aws_route_table" "private" {
   tags = "${module.private_label.tags}"
 }
 
+resource "aws_route" "additional_private" {
```
Comment (comeanother) [inline private.tf]: Rename to `private_additional`.

File: public.tf [hunk 4]
```diff
@@ -29,27 +29,30 @@ resource "aws_subnet" "public" {
...
-  subnet_id      = "${element(aws_subnet.public.*.id, count.index)}"
-  route_table_id = "${aws_route_table.public.id}"
+resource "aws_route" "default_public" {
+  count                  = "${length(var.availability_zones)}"
+  route_table_id         = "${element(aws_route_table.public.*.id, count.index)}"
+  destination_cidr_block = "0.0.0.0/0"
+  gateway_id             = "${var.igw_id}"
+}
+
+resource "aws_route" "additional_public" {
```
Comment (comeanother) [inline public.tf]: Rename to `public_additional`.

**Comment by comeanother** [comment]
Module doesn't work if we use `resource` in `additional_routes_private`.

```
  additional_routes_private = [
    {
      cidr_block = "192.168.0.0/24",
      gateway_id = "${aws_vpn_gateway.default.id}"
    },
    {
      cidr_block = "192.168.1.0/24",
      gateway_id = "${aws_vpn_gateway.default.id}"
    },
    {
      cidr_block = "192.168.2.0/24",
      gateway_id = "${aws_vpn_gateway.default.id}"
    },
    {
      cidr_block = "192.168.3.0/24",
      gateway_id = "${aws_vpn_gateway.default.id}"
    }
  ]
```

**Comment by s2504s** [comment]
Terraform apply on the test environment:
image

**Comment by s2504s** [comment]
Terraform plan after apply:
image

**Comment by aknysh** [comment]
closing it since it's very old and the module has been updated many times since then

**Pull request closed without merging**